server/service/src/invoice/outbound_return/generate_outbound_return_lines.rs

```diff
@@ -26,62 +29,178 @@ pub fn generate_outbound_return_lines(
         return_id,
     }: GenerateOutboundReturnLinesInput,
 ) -> Result<ListResult<OutboundReturnLine>, ListError> {
```

Comment:
I think this whole function could be written a lot better, but I've already spent quite a while on it. How would you improve this?